app/assets/javascripts/dradis/plugins/csv/upload.js

```diff
@@ -15,24 +15,58 @@ window.addEventListener('job-done', function(e){
 
 document.addEventListener('turbolinks:load', function() {
   if ($('body.upload.new').length) {
-    $('[data-behavior=type-select]').on('change', function() {
+    $('[data-behavior=type-select]').on('change', function(e) {
       var $nodeSelect = $('select option[value="node"]:selected').parent();
 
+      // Disable Node Label option
       $('[data-behavior=type-select]').each(function(i, select) {
         var $tr = $(select).closest('tr');
-
-        $tr.find('[data-behavior=na-field-label]').addClass('d-none');
-        $tr.find('[data-behavior=default-field-label]').removeClass('d-none');
-
         if ($nodeSelect.length && !$nodeSelect.is($(select))) {
           $(select).find('option[value="node"]').attr('disabled', 'disabled');
         } else {
           $(select).find('option[value="node"]').removeAttr('disabled');
         }
       });
 
-      $nodeSelect.closest('tr').find('[data-behavior=na-field-label]').removeClass('d-none');
-      $nodeSelect.closest('tr').find('[data-behavior=default-field-label]').addClass('d-none');
+      // Update fields column labels
+      var hasNoFields = $(e.target).val() == 'skip' || $(e.target).val() == 'node';
+      if (hasNoFields) {
+        $(e.target).closest('tr').find('[data-behavior=field-label]').text('N/A');
+      }
+      else {
+        var header = $(e.target).closest('tr').find('td:nth-child(2)').text();
```

Comment:
I think storing the header value in a data attribute, something like `data-original-text="header"` or something, is easier to understand than trying to get the text value from the 2nd `td` element in the row.

Reply:
Thanks I like this approach better